### PR TITLE
Implement autodebug function in JS library code. NFC

### PR DESCRIPTION
**Pull request by sbc100**
I needed this past of #17581, since inlining the functions into
`asmLibraryArg` like we were doing before confuses to `emitDCEGraph`
acorn pass (it ignores inline function bodies AFAICT).

### CODE REVIEW

**Comment by sbc100** [comment]
PTAL.. I need to land this before I can land #17581... I guess you probably already reviews these since #17581 already got an lgtm.

**Review by kripken — APPROVED** [review]
Nice!

**Pull request merged**
